plugins/sigma/src/parse.cpp

```diff
@@ -189,7 +189,7 @@ struct detection_parser : parser_base<detection_parser> {
 // - Regular expressions are case-sensitive by default
 // - You don't have to escape characters except the string quotation
 //   marks '
-std::optional<pattern> make_pattern(std::string_view str) {
+std::optional<caf::expected<pattern>> make_pattern(std::string_view str) {
```

Comment:
I find it clearer with both. Using unexpected with error equals none is really hard to miss on the calling site.

Reply:
I would consider it an anti-pattern, because `expected` actually comes with the ability of providing a default-constructed error.

If the goal is to upgrade to `std::expected` transparently in the future, then this is probably indeed the better API, though.

Reply:
Removing `std::optional` semantics would probably require me to split up the workflow of `make_pattern` like this: Delegate the Sigma pattern string transpiling to a separate function `transpile_pattern_string`. Instead of calling `make_pattern` I would then have to call `transpile_pattern_string` and only if the transpiled string differs from the original string, call `pattern::make()`.
Alternatively, `make_pattern` could return a `caf::error` instead of `{}` in the string equality case. That would be easier to implement, but it would create weird semantics (transpiled string equality is technically not an error).

Reply:
What we have here works, I don't see the need to spend much engineering effort improving it.